File: 2187-minimum-time-to-complete-trips/2187-minimum-time-to-complete-trips.cpp
class Solution {
public:
    long long minimumTime(vector<int>& arr, int k) {
        
        long long left =0, right=1e14,mid;
        
        while(left <= right){
            mid=(left +right)/2; 
            long long ans=0;     
            for(int i=0;i<arr.size();i++) 
                ans+= mid/arr[i];
            if(ans < k) left =mid+1;   
            else right=mid-1;       
        }
        return left;    
            
        
    }
};

File: 2187-minimum-time-to-complete-trips/2187-minimum-time-to-complete-trips.py
class Solution:
    def minimumTime(self, arr: List[int], k: int) -> int:
        left = 0
        right = k * min(arr)
        while left < right:
            mid = (left + right) // 2
            if sum((mid // x for x in arr)) >= k:
                right = mid
            else:
                left = mid + 1
        
        return left